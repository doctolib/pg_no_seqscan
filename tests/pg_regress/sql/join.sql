-- Test join query detection
-- Setup
LOAD 'pg_no_seqscan';
CREATE TABLE complex_query_foo AS (SELECT * FROM generate_series(1,10) as id);
CREATE TABLE complex_query_bar AS (SELECT * FROM generate_series(1,10) as id);
SET pg_no_seqscan.level = ERROR;

-- Test JOIN
EXPLAIN (COSTS OFF)
SELECT * FROM complex_query_foo JOIN complex_query_bar ON complex_query_foo.id = complex_query_bar.id;

SELECT * FROM complex_query_foo JOIN complex_query_bar ON complex_query_foo.id = complex_query_bar.id;

-- Cleanup
DROP TABLE complex_query_foo, test_join_bar;
