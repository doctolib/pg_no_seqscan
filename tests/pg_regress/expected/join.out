-- Test join query detection
-- Setup
LOAD 'pg_no_seqscan';
CREATE TABLE complex_query_foo AS (SELECT * FROM generate_series(1,10) as id);
CREATE TABLE complex_query_bar AS (SELECT * FROM generate_series(1,10) as id);
SET pg_no_seqscan.level = ERROR;
-- Test JOIN
EXPLAIN (COSTS OFF)
SELECT * FROM complex_query_foo JOIN complex_query_bar ON complex_query_foo.id = complex_query_bar.id;
                         QUERY PLAN                          
-------------------------------------------------------------
 Merge Join
   Merge Cond: (complex_query_foo.id = complex_query_bar.id)
   ->  Sort
         Sort Key: complex_query_foo.id
         ->  Seq Scan on complex_query_foo
   ->  Sort
         Sort Key: complex_query_bar.id
         ->  Seq Scan on complex_query_bar
(8 rows)

SELECT * FROM complex_query_foo JOIN complex_query_bar ON complex_query_foo.id = complex_query_bar.id;
ERROR:  A 'Sequential Scan' on complex_query_foo,complex_query_bar has been detected.
  - Run an EXPLAIN on your query to check the query plan.
  - Make sure the query is compatible with the existing indexes.

Query: SELECT * FROM complex_query_foo JOIN complex_query_bar ON complex_query_foo.id = complex_query_bar.id;

-- Cleanup
DROP TABLE complex_query_foo, test_join_bar;
ERROR:  table "test_join_bar" does not exist
